kernel/src/main/scala/cats/kernel/instances/DeadlineInstances.scala

```diff
@@ -0,0 +1,34 @@
+package cats.kernel
+package instances
+
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.{Deadline, FiniteDuration}
+
+trait DeadlineInstances {
+  implicit val catsKernelStdOrderForDeadline
+    : Order[Deadline] with Hash[Deadline] with LowerBounded[Deadline] with UpperBounded[Deadline] = new DeadlineOrder
+}
+
+trait DeadlineBounded extends LowerBounded[Deadline] with UpperBounded[Deadline] {
+  override def minBound: Deadline = Deadline(FiniteDuration(-Long.MaxValue, TimeUnit.NANOSECONDS))
+  override def maxBound: Deadline = Deadline(FiniteDuration(Long.MaxValue, TimeUnit.NANOSECONDS))
+}
+
+class DeadlineOrder extends Order[Deadline] with Hash[Deadline] with DeadlineBounded { self =>
```

Comment:
I think a BoundedSemilattice may also make sense here (on max?) or maybe it is unclear so we should let users make their own.

Reply:
It definitely is unclear to me how to implement that. So I'd appreciate either an approval or some help.

Reply:
The `BoundedSemilattice` is a `Monoid`, so you'd need to define `empty` and `combine`.  You also inherit these typeclasses, and therefore laws:

* a `CommutativeSemigroup`: `combine(x, y) === combine(y, x)`
* a `Band`: `combine(x, x) == x`

If you combine two deadlines, would you want the result to be the earlier deadline (`min`) or the later (`max`)?  I can imagine both, so I'm a bit skeptical of the instance.

And what should `empty` be?  `Deadline.now` is tempting, but changes on each call.  I suppose we'd use `minBound` if we combined with `max`, and `maxBound` if we combined with `min`.  Or we could escape this question by being a `Semilattice` instead of a `BoundedSemilattice`.

I think if someone can make a persuasive case for min or max being the canonical instance, we could handle this in a separate PR.

Reply:
empty would be either the minimum or maximum value depending on which BoundedSemilattice you pick. `now` would not be lawful.

Reply:
Note: we already have a constructor function for picking a max or min semilattice if you give the corresponding empty value. I agree it isn't clear if you want min or max here.

Probably some applications would want one and others the other.

Reply:
If it is dependent on the application, I would refrain from implementing it here and merging this PR.
Later one could implement the `Semilattice` in a follow-up PR anyways.

Reply:
I'm pretty well convinced the `empty` follows from `combine` and any eventual instance should be monoidal.  But there's no obvious canonical instance, and no reason to block this waiting for one.  Let's get a second approval and get this shipped.